test_util/launch.py

```diff
@@ -1,19 +1,139 @@
+        return {
+            'type': 'cloudformation',
+            'stack_name': self.config['stack_name'],
+            'provider_info': {
+                'region': self.config['region'],
+                'access_key_id': self.config['access_key_id'],
+                'secret_access_key': self.config['secret_access_key']}}
+
+    def wait(self, info):
+        # TODO: should this support the case where the cluster is being updated?
+        cf = self.get_stack(info)
+        status = cf.get_stack_details()['StackStatus']
+        if status == 'CREATE_IN_PROGRESS':
+            cf.wait_for_stack_creation(wait_for_poll_min=0)
+        elif status == 'CREATE_COMPLETE':
+            pass
+        else:
+            raise Exception('')
+
+    def describe(self, info):
+        cf = self.get_stack(info)
+        return {
+            'masters': convert_host_list(cf.get_master_ips()),
+            'private_agents': convert_host_list(cf.get_private_agent_ips()),
+            'public_agents': convert_host_list(cf.get_public_agent_ips())}
+
+    def delete(self, info):
+        self.get_stack().delete(info)
+
+    def get_stack(self, info):
+        """Returns the correct class interface depending how the AWS CF is configured
+        NOTE: only supports Simple Cloudformation currently
+        """
+        return DcosCfSimple(info['stack_name'], self.boto_wrapper)
+
+
+def get_launcher(launcher_type, provider_info):
+    """Returns a launcher given a launcher type (string) and a dictionary of
+    appropriate provier_info
+    """
+    if launcher_type == 'cloudformation':
+        assert 'region' in provider_info
+        assert 'access_key_id' in provider_info
+        assert 'secret_access_key' in provider_info
+        boto_wrapper = BotoWrapper(
+            provider_info['region'], provider_info['access_key_id'], provider_info['secret_access_key'])
+        return AwsCloudformationLauncher(boto_wrapper)
+    else:
+        raise Exception('Unsupported configuration!')
+
+
+def convert_host_list(host_list):
+    """ Makes Host tuples more readable when using describe
+    """
+    return [{'private_ip': h.private_ip, 'public_ip': h.public_ip} for h in host_list]
 
 
 def main():
-    docopt(__doc__)
+    args = docopt(__doc__, version='DC/OS Launch 1.0')
+
+    if args['create']:
+        info_path = args['--info-path']
+        assert not os.path.exists(info_path), 'There is already a cluster info at {}'.format(info_path)
+        config = yaml.load(load_string(args['--config-path']))
+        assert 'this_is_a_temporary_config_format_do_not_put_in_production' in config,\
+            'Configs used by dcos-launch must have the non-production flag'
+        write_json(info_path, get_launcher(config['type'], config['provider_info']).create())
+        sys.exit(0)
+
+    info = load_json(args['--info-path'])
+
+    if args['wait']:
+        get_launcher(info['type'], info['provider_info']).wait()
```

Comment:
Technically should also wrap pulling the fields out of the json so that we get good error messages if they aren't set / the info_json doesn't have them. (Error: `no field 'foo' in 'cluster_info.json` is a much nicer message than a python index exception)